### PR TITLE
Enhancement: store or check metanode, datanode port before start

### PR DESCRIPTION
**Pull request by ZhuZhengyi**
Signed-off-by: zhuzhengyi1 <zhuzhengyi@jd.com>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: datanode/partition_raft.go [hunk 1]
```diff
@@ -436,6 +436,16 @@ func (s *DataNode) startRaftServer(cfg *config.Config) (err error) {
 
 	s.parseRaftConfig(cfg)
 
+	constCfg := config.ConstConfig{
+		Listen:           s.port,
+		RaftHeartbetPort: s.raftHeartbeat,
+		RaftReplicaPort:  s.raftReplica,
+	}
+	if ok, err1 := config.CheckOrStoreConstCfg(s.raftDir, config.DefaultConstConfigFile, &constCfg); !ok {
```
Comment (mervinkid) [inline datanode/partition_raft.go:444]: Do not use numbers when naming variables.

File: metanode/metanode.go [hunk 2]
```diff
@@ -195,6 +195,16 @@ func (m *MetaNode) parseConfig(cfg *config.Config) (err error) {
 		return fmt.Errorf("bad cfgRaftReplicaPort config")
 	}
 
+	constCfg := config.ConstConfig{
+		Listen:           m.listen,
+		RaftHeartbetPort: m.raftHeartbeatPort,
+		RaftReplicaPort:  m.raftReplicatePort,
+	}
+	if ok, err1 := config.CheckOrStoreConstCfg(m.metadataDir, config.DefaultConstConfigFile, &constCfg); !ok {
```
Comment (mervinkid) [inline metanode/metanode.go:203]: Do not use numbers when naming variables.

File: datanode/partition_raft.go (hunk 1 above)
Comment (ZhuZhengyi) [reply datanode/partition_raft.go]: ok

File: metanode/metanode.go (hunk 2 above)
Comment (ZhuZhengyi) [reply metanode/metanode.go]: ok

**Pull request merged**